server/src/main/java/org/elasticsearch/plugins/PluginsService.java

```diff
@@ -57,6 +58,14 @@
 
 public class PluginsService implements ReportingService<PluginsAndModules> {
 
+    /**
+     * A loaded plugin is one for which Elasticsearch has successfully constructed an instance of the plugin's class
+     * @param info Metadata about the plugin, usually loaded from plugin properties
+     * @param instance The constructed instance of the plugin's main class
+     * @param loader   The classloader for the plugin
+     */
+    record LoadedPlugin(PluginInfo info, Plugin instance, @Nullable ClassLoader loader) {}
```

Comment:
I don't think we need to use Nullable. In production this should not be null, we just have tests that abuse this contract, knowing how it is used.